src/main/scala/apps/harrisCornerDetection.scala

```diff
@@ -301,11 +302,11 @@ object harrisCornerDetection {
 
   object NoPipe {
     def create: NoPipe = NoPipe(
-      gen.OpenCLKernel(sobelX),
-      gen.OpenCLKernel(sobelY),
-      gen.OpenCLKernel(mul),
-      gen.OpenCLKernel(gaussian),
-      gen.OpenCLKernel(coarsity)
+      gen.opencl.kernel.fromExpr("sobelX")(sobelX),
+      gen.opencl.kernel.fromExpr("sobelY")(sobelY),
+      gen.opencl.kernel.fromExpr("mul")(mul),
+      gen.opencl.kernel.fromExpr("gaussian")(gaussian),
+      gen.opencl.kernel.fromExpr("coarsity")(coarsity)
```

Comment:
I have adopted your suggestions and changed the API so that the kernel/function name (and the work group config) are now (optionally) given early. So the calls above now look like:
```
gen.opencl.kernel("sobelY").fromExpr(sobelY)
```
or
```
gen.opencl.kernel.fromExpr(sobelY)
```
when using a default name.